planetiler-core/src/main/java/com/onthegomap/planetiler/expression/Expression.java

```diff
@@ -359,11 +387,18 @@ public boolean evaluate(SourceFeature input, List<String> matchKeys) {
 
     @Override
     public String toString() {
-      return "matchAny(" + Format.quote(field) + ", " + values.stream().map(Format::quote)
+      return "matchAny(" + Format.quote(field) + ", " + values.stream().map(Expression::stringifyTypedObject)
         .collect(Collectors.joining(", ")) + ")";
     }
   }
 
+  private static String stringifyTypedObject(Object o) {
+    if (o instanceof String) {
+      return Format.quote(o.toString());
+    }
+    return o.toString();
+  }
```

Comment:
`stringifyTypedObject` loses the numeric type, and `Integer.valueOf(1)` is not equal to `Long.valueOf(1)` so it might be safter to do an instanceof check for a list of supported types and emit strings like `6L` that encode the right numeric type.

_But..._ I also realize that we don't use values anywhere other than comparing string representations so maybe it's fine to just encode everything as a string?